controllers/rollingupgrade_controller.go

```diff
@@ -259,11 +259,10 @@ func (r *RollingUpgradeReconciler) WaitForDesiredInstances(ruObj *upgrademgrv1al
 			return err
 		}
 
-		val, ok := r.ruObjNameToASG.Load(ruObj.Name)
-		if !ok {
+		asg, err := r.GetAutoScalingGroup(ruObj.Name)
+		if err != nil {
 			return fmt.Errorf("Unable to load ASG with name: %s", ruObj.Name)
 		}
-		asg := val.(*autoscaling.Group)
```

Comment:
nice clean-up, after typing this block out a few times, I should have made into fn myself